src/stactools/package/commands.py

```diff
@@ -0,0 +1,68 @@
+import click
+import logging
+import json
+import os
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+def read_metadata_json(source: str) -> dict:
+    """Collect a dict from a local or remote json source.
+
+    Enables the flexible use of json metadata files in CLI commands.
+
+    Args:
+        source (str): Path or url to a .json file
+
+    Returns:
+        dict: The contents of the json file converted to a dictionary
+    """
+    if os.path.isfile(source):
+        with open(source) as f:
+            data = json.load(f)
+    else:
+        metadata_response = requests.get(source)
+        data = metadata_response.json()
```

Comment:
Since we have `fsspec`, can we just use `with fsspec.open` and let it do the work for us?